sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -796,8 +807,9 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     val schemaJson = CatalogTable.readLargeTableProp(table.properties, DATASOURCE_SCHEMA)
     if (schemaJson.isDefined) {
       val schemaFromTableProps = DataType.fromJson(schemaJson.get).asInstanceOf[StructType]
+      val rawSchema = CharVarcharUtils.getRawSchema(schemaFromTableProps)
```

Comment:
Maybe we should create a method `getSchemaFromTableProperties` and call it in these places.

Reply:
done